core/controllers/editor.py

```diff
@@ -613,59 +613,83 @@ class ImageUploadHandler(EditorHandler):
     # The string to prefix to the filename (before tacking the whole thing on
     # to the end of 'assets/').
     _FILENAME_PREFIX = 'image'
+    _decorator = None
 
-    @acl_decorators.can_edit_exploration
-    def post(self, exploration_id):
+    def post(self, entity_type, entity_id):
         """Saves an image uploaded by a content creator."""
 
-        raw = self.request.get('image')
-        filename = self.payload.get('filename')
-        if not raw:
-            raise self.InvalidInputException('No image supplied')
-
-        allowed_formats = ', '.join(
-            feconf.ACCEPTED_IMAGE_FORMATS_AND_EXTENSIONS.keys())
-
-        # Verify that the data is recognized as an image.
-        file_format = imghdr.what(None, h=raw)
-        if file_format not in feconf.ACCEPTED_IMAGE_FORMATS_AND_EXTENSIONS:
-            raise self.InvalidInputException('Image not recognized')
-
-        # Verify that the file type matches the supplied extension.
-        if not filename:
-            raise self.InvalidInputException('No filename supplied')
-        if filename.rfind('.') == 0:
-            raise self.InvalidInputException('Invalid filename')
-        if '/' in filename or '..' in filename:
-            raise self.InvalidInputException(
-                'Filenames should not include slashes (/) or consecutive dot '
-                'characters.')
-        if '.' not in filename:
-            raise self.InvalidInputException(
-                'Image filename with no extension: it should have '
-                'one of the following extensions: %s.' % allowed_formats)
+        if entity_type == fs_domain.ENTITY_TYPE_EXPLORATION:
```

Comment:
@aks681, this is brilliant! It's really clear and I think it's a great solution.

 @anubhavsinha98, you can perhaps adopt a similar convention for your PR.